### PR TITLE
feat: adding UpdateStateOnMisbehaviour to ClientState interface

### PR DESCRIPTION
**Pull request by seantking**
## Description

closes: #1193 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/core/exported/client.go [hunk 1]
```diff
@@ -56,8 +56,10 @@ type ClientState interface {
 	// Genesis function
 	ExportMetadata(sdk.KVStore) []GenesisMetadata
 
-	// Update and Misbehaviour functions
+	// UpdateStateOnMisbehaviour updates state upon misbehaviour
```
Comment (colin-axner) [inline modules/core/exported/client.go:59]:
```suggestion
	// UpdateStateOnMisbehaviour should perform appropriate state changes on a client state given that misbehaviour has been detected and verified
```

File: modules/light-clients/06-solomachine/types/update.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ func (cs ClientState) CheckHeaderAndUpdateState(
 
 	foundMisbehaviour := cs.CheckForMisbehaviour(ctx, cdc, clientStore, msg)
 	if foundMisbehaviour {
-		return cs.UpdateStateOnMisbehaviour(ctx, cdc, clientStore)
```
Comment (colin-axner) [inline modules/light-clients/06-solomachine/types/update.go:29]: ? 

Removing the return causes `cs.UpdateState` to be called as well

Comment (seantking) [reply modules/light-clients/06-solomachine/types/update.go:29]: Indeed. I was just being lazy and moving fast. I figured we want to remove this entire `CheckHeaderAndUpdateState` function in the next batch of PRs anyway but on second glance there's no reason to not have it in. I've added the correct return statement.

**Review by damiannolan — APPROVED** [review]

File: modules/light-clients/06-solomachine/types/update.go [hunk 3]
```diff
@@ -26,7 +26,8 @@ func (cs ClientState) CheckHeaderAndUpdateState(
...
+		cs.UpdateStateOnMisbehaviour(ctx, cdc, clientStore)
```
Comment (colin-axner) [inline modules/light-clients/06-solomachine/types/update.go:29]: I guess this function will be deleted soon enough

Comment (colin-axner) [reply modules/light-clients/06-solomachine/types/update.go:29]: ah nvm! I see the return right below now!

**Pull request merged**